### DISCUSSION TITLE
Compiling updated changes into flashable bin files?

### DISCUSSION THREAD

**Discussion by Helder1981** [konwektor/BlueRetro#7] [General]
Hi I was wondering if there is a guide or how to for compiling and flashing the changes made to bin files? I have spent 2 days following various guides and tutorials for installing ESP-IDF and they all fail one way or another on windows. 

I am assuming you are not using windows as there is no easy working solution I have found for doing so. Could you explain how you set up your ESP-IDF and what OS you were using as I'd like to make changes to the code for the system it emulates and then compile it so it's easily sharable.

thank you if you can help.

**Comment by konwektor** [comment]
I'm using esp-idf under Windows. [https://dl.espressif.com/dl/esp-idf/](url).
Download online installer   and make full install of all components. Do it with asministrator privileges. Mark option for having shortcuts on desktop (one is power shell shortcut, second command line short) After installation start esp-idf with created link, but also with admin privilegee(right Mouse button). Use Cmd one(Better not playin with power shell, as Admin). Go to your project, or clone one from git, and start compilation."idf.py build".  
You need alwas starting esp-idf with admin privileges or could be problems with system paths.  

I got also one extra folder with git clones of different esp-idf versions. From v4.0 to v5.0.1
. Sometimes git clones doesnt work and dont have any info about  idf version for compilation.
For change aktive esp-idf framework:
I start cmd.exe (admin priv), goin into choosen idf version,,and inside folder are 2 important files. Install.bat and export.bat. run both.
Install downloading missing files for this specifed version, and replacing with those already installed.
Export adds present idf folder to system path. This way changing idf version is quite fast.
Dont forget about :
 "Limitations: - The installation path of ESP-IDF and ESP-IDF Tools must not be longer than 90 characters. Too long installation paths might result in a failed build. - The installation path of Python or ESP-IDF must not contain white spaces or parentheses. - The installation path of Python or ESP-IDF should not contain special characters (non-ASCII) unless the operating system is configured with “Unicode UTF-8” support"

I hope , its wright now enough clear to You.
P.S.sorry for so late answer, I didnt  Look before in discussions.
